src/main/java/core/basesyntax/dao/jdbc/CarDaoJdbcImpl.java

```diff
@@ -0,0 +1,206 @@
+package core.basesyntax.dao.jdbc;
+
+import core.basesyntax.dao.CarDao;
+import core.basesyntax.exception.DataProcessingException;
+import core.basesyntax.lib.Dao;
+import core.basesyntax.model.Car;
+import core.basesyntax.model.Driver;
+import core.basesyntax.model.Manufacturer;
+import core.basesyntax.util.ConnectionUtil;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Dao
+public class CarDaoJdbcImpl implements CarDao {
+    @Override
+    public Car create(Car car) {
+        String queryInsert = "INSERT INTO cars (model, manufacturer_id) "
+                + "VALUES (?, ?)";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement stInsert = connection.prepareStatement(queryInsert,
+                        Statement.RETURN_GENERATED_KEYS)) {
+            stInsert.setString(1, car.getModel());
+            stInsert.setLong(2, car.getManufacturer().getId());
+            stInsert.execute();
+            ResultSet resultSet = stInsert.getGeneratedKeys();
+            if (resultSet.next()) {
+                car.setId(resultSet.getObject("GENERATED_KEY", Long.class));
+            }
+            return car;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't insert car " + car, e);
+        }
+    }
+
+    @Override
+    public Optional<Car> get(Long id) {
+        String querySelect = "SELECT c.id, c.model, c.manufacturer_id, cd.driver_id FROM cars c "
+                + "INNER JOIN manufacturers m ON m.id=c.manufacturer_id "
+                + "INNER JOIN cars_drivers cd ON cd.car_id=c.id "
+                + "INNER JOIN drivers d ON d.id=cd.driver_id "
+                + "WHERE c.deleted=FALSE AND d.deleted=FALSE AND c.id=?";
```

Comment:
in methods get(), getAll() and getAllByDriver() we should have one query fetching info about car and its manufacturer (1 join needed)

and the second query getting info about all the drivers of the car (separate connection in a separate method)